src/Pages/GlobalConfigurations/BuildInfra/services.tsx

```diff
@@ -0,0 +1,183 @@
+            // Removing id from it since we do not have a configuration
+            // While generating payload, we will check on the basis of id and active flag.
+            acc[key] = {
+                key,
+                value: defaultConfiguration.value,
+                // saving profile name as undefined and this would be a check, if we are deriving from default or not
+                unit: defaultConfiguration.unit,
+                active: false,
+            }
+        }
+        acc[key].defaultValue = {
+            value: defaultConfiguration.value,
+            unit: defaultConfiguration.unit,
+        }
+        return acc
+    }, {} as BuildInfraConfigurationMapType)
+
+    return {
+        configurationUnits,
+        profile: {
+            ...(profile && getBaseProfileObject(fromCreateView, profile)),
+            configurations,
+        },
+    }
+}
+
+export const getBuildInfraProfileByName = async ({
+    name,
+    fromCreateView,
+}: GetBuildInfraProfileType): Promise<BuildInfraProfileResponseType> => {
+    try {
+        const response = await get(`${ROUTES.INFRA_CONFIG_PROFILE}/${name}`)
+        const {
+            result: { configurationUnits, defaultConfigurations, profile },
+        } = response as ResponseType<BuildInfraProfileAPIResponseType>
+
+        return getTransformedBuildInfraProfileResponse({
+            configurationUnits,
+            defaultConfigurations,
+            profile,
+            fromCreateView,
+        })
+    } catch (error) {
+        showError(error)
+        throw error
+    }
+}
+
+const getBuildInfraProfilePayload = (
+    profileInput: CreateBuildInfraProfileType['profileInput'],
+): CreateBuildInfraServicePayloadType => {
+    const currentConfigurations = profileInput.configurations
+    const configurationKeys = Object.keys(currentConfigurations) as BuildInfraConfigTypes[]
+    // would only keep the configurations with id or active flag as true
+    const configurations: CreateBuildInfraServiceConfigurationType[] = configurationKeys.reduce((acc, key) => {
+        const configuration = currentConfigurations[key]
+        if (configuration.id || configuration.active) {
+            acc.push({
+                id: configuration.id,
+                key,
+                value: Number(configuration.value),
+                unit: configuration.unit,
+                active: configuration.active,
+            })
+        }
+        return acc
+    }, [] as CreateBuildInfraServiceConfigurationType[])
+
+    const payload: CreateBuildInfraServicePayloadType = {
+        name: profileInput.name,
+        description: profileInput.description,
+        type: profileInput.type,
+        configurations,
+    }
+    return payload
+}
+
+export const updateBuildInfraProfile = ({ name, profileInput }: UpdateBuildInfraProfileType) => {
+    const payload = getBuildInfraProfilePayload(profileInput)
+    // TODO: Would remove this
+    console.log(payload, 'edit payload')
```

Comment:
Please remove this and other console.log

Reply:
Done